platform_plugin_ontask/data_summary/backends/grade.py

```diff
@@ -0,0 +1,77 @@
+"""Data summary for grade data."""
+
+from collections import defaultdict
+
+from opaque_keys.edx.keys import CourseKey
+
+from platform_plugin_ontask.data_summary.backends.base import DataSummary
+from platform_plugin_ontask.edxapp_wrapper.courseware import get_score
+from platform_plugin_ontask.edxapp_wrapper.enrollments import get_user_enrollments
+from platform_plugin_ontask.utils import get_course_components
+
+
+class ComponentGradeDataSummary(DataSummary):
+    """
+    Data summary for the grade of the components in a course.
+
+    A component grade data summary is a summary of all the grade data of the
+    components in a specific course. Each record contains the user ID and the
+    grade of each component.
+
+    `get_data_summary` performs the following steps:
+
+    1. Get the course key from the course ID.
+    2. Get all the enrollments for the course.
+    3. Get all the course components for the course.
+    4. Create a dictionary with the component grade data summary.
+
+    Example result:
+
+    ```python
+
+    grade_data_summary = ComponentGradeDataSummary()
+    grade_data_summary.get_data_summary()
+
+    {
+        "user_id": {
+            "0": 5,
+            "1": 6,
+        },
+        "component_9c56dbeb30504c8fb799553f080cf15d_grade": {
+            "0": 1,
+            "1": 0,
+        },
+        "component_6c7e4b1b7b7e4b3e8b7b7e4b3e8b7b7_grade": {
+            "0": 0,
+            "1": 1,
+        },
+        ...
+    }
+
+    ```
+    """
+
+    GRADE = "component_{}_grade"
```

Comment:
```suggestion
    GRADE_COLUMN_NAME = "component_{}_grade"
```

Reply:
This, and the other style comments are non-blocking.

Reply:
Done: https://github.com/eduNEXT/platform-plugin-ontask/commit/1b3fe5c5c597068413a555edf47ff722223e2b02